modules/dns-helper/dns_local_server.go

```diff
@@ -0,0 +1,112 @@
+package dns_helper
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/miekg/dns"
+)
+
+var testDomain = "gruntwork.io"
+
+// DNSData type
+type DNSData map[DNSQuery][]DNSAnswer
+
+func setupTestDNSServers() (ns1, ns2 string, dnsData1, dnsData2 *DNSData) {
```

Comment:
You don't need to use pointers with maps, because they are passed by reference. [Source](https://blog.golang.org/maps):

> Map types are reference types, like pointers or slices

Reply:
Refactored the whole DNSDatabase concept.